### ISSUE TITLE
[HOLD for payment 2022-09-16] [$4000] There's a very small white gap at bottom for TextInput when using autofill - @rushatgabhane

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#7470]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open app
2. Use autofill on the login screen 
3. Navigate to Settings>Payments>Add Debit card 
4. Use autofill 

## Expected Result:
Autofill should match the box 

## Actual Result:
There's a very small white gap at bottom for TextInput when using autofill
There is Discovered while testing this PR #7099 

## Workaround:
unknown. 

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.33-0
**Reproducible in staging?:** Y  
**Reproducible in production?:** Y 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 
![image - 2022-01-30T224559 026](https://user-images.githubusercontent.com/43995119/151736102-cf6853f7-9152-41c3-b99a-288580541016.png) 
![image - 2022-01-30T224600 550](https://user-images.githubusercontent.com/43995119/151736113-1b63ff70-6152-49f8-b4e6-43e3cc2c107a.png)

**Upwork job URL:** https://www.upwork.com/jobs/~0132057b0680124108
**Issue reported by:** @rushatgabhane
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1642714377248800

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @francoisl (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @trjExpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @rushatgabhane (`Exported`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @stitesExpensify (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by okansahin** [comment]
I read the issue and contributing guide.

I logged to app(https://new.expensify.com/) to facing issue but there is no option to add debit card for me. And in mobile there is no any options to add payment method. Maybe its about my region.
<img width="1552" alt="Screen Shot 2022-01-31 at 22 43 44" src="https://user-images.githubusercontent.com/1966034/151861633-7978f355-a026-4dab-8b33-befeaee9adbf.png">

Therefore, I could not examine it with the inspect tool.

But first thing came to my mind was try to change background color completely with input:-webkit-autofill.

If I can make the gap visible, I believe other ideas will come to my mind. :)

**Comment by rushatgabhane** [comment]
@okansahin I reckon debit card is hidden under a beta. I was able to access it on main though

**Comment by okansahin** [comment]
> @okansahin I reckon debit card is hidden under a beta. I was able to access it on main though

Hello, 

### Web

I read contributing guide and tried to got issue on desktop and mobile but there is no option "Add debit card" in payment settings on https://new.expensify.com . I don't know maybe it's relational with my region. In mobile view of chrome, no any option.

<img width="1552" alt="Screen Shot 2022-01-31 at 23 14 35" src="https://user-images.githubusercontent.com/1966034/151865913-a864d9be-247e-49af-bdfc-be3f5bbce250.png">
![IMG_CF4235E8FADC-1](https://user-images.githubusercontent.com/1966034/151867251-adc196e7-7977-4d01-98f0-dab1a95c1ae7.jpeg)

The first thing came in my mind is try to change completely change auto fill background with input:-webkit-autofill selector for web.
If I make the problem visible I believe I can produce any other ideas.

If I have an error my flow, please correct me to reproducing the problem.

### React Native,

I tried to run the project in simulator. But It gives me missing package error react-native-blob-util, after I added it, then gives me the error on the below image. I know hermes engine is for android  🤔

and after @rushatgabhane's reply I tried on production I react the "Add Bank Card" but there is no gap for my autofill on zipcode, I can't tried the others because I don't have saved card.

And for password on login with autofill,
<img width="460" alt="Screen Shot 2022-02-01 at 00 02 03" src="https://user-images.githubusercontent.com/1966034/151872768-5ab73e28-47cc-4e7a-8fc6-cd4bf7d24fa3.png">

As a result, more time is required to make the problem visible. I believe that I can easily get used to the project. I am familiar with the tools and methods used in this project.

Thank you

**Comment by stitesExpensify** [comment]
@okansahin as @rushatgabhane mentioned, it's probably a beta issue. Try updating [this line](https://github.com/Expensify/App/blob/50ebc74b906bc9ae01b9f52d301d672b53e29cbd/src/libs/Permissions.js#L11) to always return true and see if you can see the debit card option on the dev build. 

With regards to your iOS build I would recommend reaching out in the private slack room.

**Comment by okansahin** [comment]
Hello,

I run the project. It was my fault with my environment, I was a little hasty. It's okay now.

But I still can't see the problem. Someone can tell me exact system(browser, os etc.) which is this gap is visible. Or Is this issue fixed?

@stitesExpensify Can you please send me invite to Slack? I'm interested and want to fix this.

#### Safari on iOS Simulator

![Screen Shot 2022-02-01 at 16 04 39](https://user-images.githubusercontent.com/1966034/151975809-afa80857-c3ff-4958-bfa3-6d95b50119d8.png)

![Screen Shot 2022-02-01 at 16 08 30](https://user-images.githubusercontent.com/1966034/151975839-7dd07a33-2fdf-4d67-8c2e-312a1b369cd1.png)

![Screen Shot 2022-02-01 at 16 08 36](https://user-images.githubusercontent.com/1966034/151975856-ce01cdb4-4c45-4c86-980a-e908e055449b.png)

#### Safari on MacOS

![Screen Shot 2022-02-01 at 16 12 56](https://user-images.githubusercontent.com/1966034/151975912-3f5d207a-f547-4d58-9b33-19fee9b8780a.png)

#### Chrome on MacOS

![Screen Shot 2022-02-01 at 16 11 09](https://user-images.githubusercontent.com/1966034/151975971-5acc97f6-bc02-4f02-ae42-cb04dbb359a4.png)

![Screen Shot 2022-02-01 at 16 11 25](https://user-images.githubusercontent.com/1966034/151976000-0e1c62fd-0acc-4ffb-b7e5-163795afd085.png)

**Comment by rushatgabhane** [comment]
Will reply in a bit

**Comment by rushatgabhane** [comment]
@okansahin try the in built autofill on Android for the password page. It is the most easily visible.

> Can you please send me invite to Slack? I'm interested and want to fix this.

To join #expensify-open-source, just email contributors@expensify.com with the subject `Slack Channel Invite` and include a link to your Upwork profile.

**Comment by trjExpensify** [comment]
@okansahin are you planning to submit a proposal for this job, and if so, do you want to then apply for it on Upwork [here](https://www.upwork.com/jobs/~01909f315eb8bc7426)?

**Comment by okansahin** [comment]
@trjExpensify I reproduce this gap only on android's login screen(it's not reproducible in adding payment card screen) when password is visible. If you confirm that, I will try to find a way. Otherwise could you give me exact version/device which this gap is reproducible?

**Comment by rushatgabhane** [comment]
@okansahin cool, I'll update you on that within 30 mins

**Comment by rushatgabhane** [comment]
> it's not reproducible in adding payment card screen

@okansahin I agree.

**Comment by okansahin** [comment]
Thank you @rushatgabhane 

I will look up to it, and let you know that I can fix or not.

**Comment by stitesExpensify** [comment]
Hi @okansahin any update here?

**Comment by okansahin** [comment]
Hello @stitesExpensify sorry for delay. I was busy with my other tasks.

> > it's not reproducible in adding payment card screen
> 
> @okansahin I agree. Feel free to submit a proposal for the Android issue.

@rushatgabhane when I reviewed the DOM I detected this gap occurs in Add Debit Card screen too. If you want to reproduce this everytime, please set password input visible before autofill.

The real problem is line-height. BaseTextInput's default line-height is 20, I set it to 25 to seeing this clearly. Somehow android's autofill feature bypasses the line height if secureTextEntry is true. You can see in video below that what I'm talking about.


https://user-images.githubusercontent.com/1966034/154512539-7a246130-3ad1-4a67-8175-6bae66ae2961.mov

I tested this actions in Add Debit Card too, and the results is same.

### PROPOSAL
I don't know why coders use lineHeight in their projects but I never use it and I never had a problem while I apply a design to frontend because I don't use lineHeight, the paddings and font size are enough I think. But I might be miss something, I would be happy if someone could teach me otherwise.

My suggestion is bypassing the lineHeight in inputStyles if passwordHidden state is true in BaseTextInput. 

Here is a test result without lineHeight property.


https://user-images.githubusercontent.com/1966034/154514893-2808011a-cdd6-46f6-8f7e-7838d294c7da.mov


Thank you

**Comment by okansahin** [comment]
Any review on my proposal? Because I have free time right now and want to seize on time, if there is any trouble. @stitesExpensify @rushatgabhane

**Comment by rushatgabhane** [comment]
> The real problem is line-height

@okansahin That makes sense to me, and thanks for the exact repro steps.
 
> My suggestion is bypassing the lineHeight in inputStyles if passwordHidden state is false in BaseTextInput.

That seems like a hacky fix. Why not get rid of lineHeight entirely?

**Comment by okansahin** [comment]
@rushatgabhane Yes! Actually, my suggestion is this. But I didn't suggest it because you could tell me that it will affect so many TextInputs. I can figure out which views could be affect and set it with padding way and test and show all affected TextInputs.

**Comment by rushatgabhane** [comment]
Actually, that might not be a good idea. source - https://github.com/Expensify/App/issues/7541#issuecomment-1049365558

**Comment by okansahin** [comment]
@rushatgabhane So then how about my proposal?

**Comment by trjExpensify** [comment]
Any update here for @okansahin, @rushatgabhane?

The Upwork job expired, so here's the new one: https://www.upwork.com/jobs/~0176b58cfaaf71ab72

**Comment by vignbenjamin435** [comment]
I have checked the upwork job post description..
I can say that i need to know which library this project is using now? And if the styled-component was used, it can be fixed by adjusting the css.
If this input field is using another library something like native-base, material UI, I need to check the Input field component.

**Comment by rushatgabhane** [comment]
> So then how about my first proposal?

I don't think we should go ahead with a hack

**Comment by rushatgabhane** [comment]
Hey @vignbenjamin435, we'd want this issue to be fixed without any extra libs. Yes, feel free to post a proposal which uses some css.

**Comment by vignbenjamin435** [comment]
Hi Rusat.
thank you for your message.
I agree with your suggestion. I can fix this problem without using any
extra libs
I can fix this problem using css.

**Comment by rushatgabhane** [comment]
@vignbenjamin435 alright, excited for your proposal.

Make sure that you read [contributing.md](https://github.com/Expensify/App/blob/317cce57c2853eed5757b4b842581828a0648fb5/CONTRIBUTING.md)  
before posting a proposal

**Comment by vignbenjamin435** [comment]
Ok. Now i am checking it seriously😊.. After i check, i will let you know.

**Comment by vignbenjamin435** [comment]
I have checked all documents....Just now i am willing to contribue..
From now on, I am going to work on your project full time.. I am full time
freelancer

**Comment by rushatgabhane** [comment]
>Do you have any suggestions for me to write an acceptable offer?

@okansahin fixing this using some css would be nice

**Comment by okansahin** [comment]
> 
> @okansahin fixing this using some css would be nice

I already do that with CSS. A class which resets lineHeight will be applied to TextInput if it's is password visible secure input. The JS part just a conditional classname.

**Comment by trjExpensify** [comment]
👋 @stitesExpensify @rushatgabhane, where are we at with the path forward for this issue?

**Comment by rushatgabhane** [comment]
sorry, I don't 100% like any proposals here. Maybe @stitesExpensify has a different opinion

**Comment by okansahin** [comment]
@rushatgabhane How kind proposals do you like 😀 I found the source of the problem which I think was the main issue, and sent a proposal like you said with css only. But you don't like, it's weird.

**Comment by rushatgabhane** [comment]
@okansahin I'm sorry that I didn't communicate with you properly here. 

We have many text inputs that can be autofilled. How will you fix it for inputs like debit card. We can't use your proposal for it. Please correct me if I'm wrong :)

**Comment by okansahin** [comment]
> > 
> > @okansahin fixing this using some css would be nice
> 
> I already do that with CSS. A class which resets lineHeight will be applied to TextInput if it's password is visible secure input. The JS part is just a conditional classname.

@rushatgabhane I think you didnt read this. As I said before the line height reset class will be applied only for secure inputs in @android. So it's not relational the other inputs.

**Comment by rushatgabhane** [comment]
Sorry if I'm missing something, but as stated in the issue, we also need to fix it for inputs in debit card page. https://github.com/Expensify/App/issues/7470#issue-1118937295
Those inputs don't have secureTextEntry, so your proposal won't fix it right?

**Comment by stitesExpensify** [comment]
@okansahin any update?

**Comment by stitesExpensify** [comment]
@okansahin bump

**Comment by stitesExpensify** [comment]
@trjExpensify I think it's time to double this issue, doesn't look like any proposals are moving forward here.

**Comment by trjExpensify** [comment]
Doubled the Upwork job to $500 here: https://www.upwork.com/jobs/~0132057b0680124108

**Comment by rushatgabhane** [comment]
@Expensify/contributor-plus feel free to pick this up. Unassigning because I'll be OOO.

There are no untriaged proposals here.

**Comment by trjExpensify** [comment]
Due a double up today in Upwork. Actioned on that [here](https://www.upwork.com/jobs/~0132057b0680124108).

**Comment by rushatgabhane** [comment]
@michaelhaxhiu oops 😂

**Comment by WelbexIT** [comment]
Hi! We solved your issue, the solution was sent to Upwork

**Comment by mananjadhav** [comment]
Went through the history, waiting for proposals.

**Comment by trjExpensify** [comment]
Doubled the price on Upwork [here](https://www.upwork.com/jobs/~0132057b0680124108)

**Comment by trjExpensify** [comment]
Doubled!

**Comment by WelbexIT** [comment]
Hi! Uploaded our solution, waiting for approval. Sent proposal to Upwork

**Comment by mananjadhav** [comment]
Hi, @WelbexIT,
Please take a look at the [contribution guidelines](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md). You need to send a proposal here, with how you plan to solve this with the technical details. One of the team members will review it here in Github and once approved, you'll be hired via Upwork to start working on it.

**Comment by jayeshmangwani** [comment]
Proposal:

This issue is happening because text input has a fixed height and we are adding component to more the defined height,

https://github.com/Expensify/App/blob/5ab06b4efc77fb36c3aa7993f70ed96e7f31d65d/src/components/TextInput/BaseTextInput.js#L206-L211
Here we passing height = componentHeightLarge = 52  value to the parent, and child we are adding borderwidth = 1 , so total (2px will be minus from the 52)
so text input height will be 50
(I have logged height using on layout and this is the response for parent and child,
parent = 51.80952453613281
child = 50.28571319580078),

https://github.com/Expensify/App/blob/5ab06b4efc77fb36c3aa7993f70ed96e7f31d65d/src/styles/styles.js#L716-L726
Here we are total using 51 height (top + bottom + lineheight, 23 + 8 + 20),thats why textinput goes little bit up from the bottom due to paddingBottom 8, reduce it by 2 and component looks fine, but I am not sure this is the right solution,


What i think best is we can add on layout to parent of the textinput component, and use that height for the textinput

Proposed changes

1.https://github.com/Expensify/App/blob/5ab06b4efc77fb36c3aa7993f70ed96e7f31d65d/src/components/TextInput/BaseTextInput.js#L16
import variables from '../../styles/variables';


2.https://github.com/Expensify/App/blob/5ab06b4efc77fb36c3aa7993f70ed96e7f31d65d/src/components/TextInput/BaseTextInput.js#L25-L35
height: variables.componentSizeLarge,

3.https://github.com/Expensify/App/blob/5ab06b4efc77fb36c3aa7993f70ed96e7f31d65d/src/components/TextInput/BaseTextInput.js#L205-L211
onLayout={event => this.setState({height: event.nativeEvent.layout.height})}


4.

https://github.com/Expensify/App/blob/5ab06b4efc77fb36c3aa7993f70ed96e7f31d65d/src/components/TextInput/BaseTextInput.js#L259-L266
{ height: this.state.height },


Result:


https://user-images.githubusercontent.com/35371050/168695082-ea616517-2810-4a03-afbe-eb8dfad6bb7b.mp4

![Screenshot_20220517-035348_New Expensify](https://user-images.githubusercontent.com/35371050/168695664-ecf8cad6-705a-45db-b0f4-b34840cac965.jpg)

**Comment by mananjadhav** [comment]
Reviewing @jayeshmangwani's proposal.

@WelbexIT are you keen on submitting the proposal here?

**Comment by mananjadhav** [comment]
I checked @jayeshmangwani's proposal and for the lack of any better proposal, I feel we can go ahead with this.

@stitesExpensify do you see any challenges with `setLayout`?

Note that I am not able to reproduce this on Web and iOS (Android build isn't working for me). So @mvtglobally are you able to reproduce this on all platforms as mentioned in the GH body.

**Comment by trjExpensify** [comment]
Not overdue, Melv. Next steps is for @stitesExpensify to review the proposal. 👍

**Comment by trjExpensify** [comment]
Same again, Melvin. @stitesExpensify can you address [this please](https://github.com/Expensify/App/issues/7470#issuecomment-1132470341), so we can move the proposal forward?

**Comment by stitesExpensify** [comment]
:wave: Apologies for the delay I was on vacation and then got covid so I've been OOO for a long time 😬 

I think that @jayeshmangwani proposal looks good! Happy to move forward with it

**Comment by mananjadhav** [comment]
@stitesExpensify Adding a proper confirmation, we're going ahead with @jayeshmangwani's proposal.

🎀 👀 🎀
C+ reviewed

**Comment by jayeshmangwani** [comment]
@mananjadhav @stitesExpensify [PR](https://github.com/Expensify/App/pull/9294) has been raised

**Comment by mananjadhav** [comment]
I should’ve commented on the PR. I am in process of reviewing your PR only and testing it. I’ll update by eod today.

**Comment by thesahindia** [comment]
I tried the solution but it didn't fix the issue on mWeb also we should fix https://github.com/Expensify/App/issues/8981 here

<img width="280" alt="Screenshot 2022-06-05 at 10 20 20 PM" src="https://user-images.githubusercontent.com/83179295/172061403-66520eb8-a1bc-4d1a-a7df-4820c9e4ba8b.png">

**Comment by stitesExpensify** [comment]
@jayeshmangwani just wanted to make sure you see this ^

**Comment by jayeshmangwani** [comment]
@stitesExpensify I am not sure but are you saying that I need to check this https://github.com/Expensify/App/issues/7470#issuecomment-1146848162 ?

**Comment by mananjadhav** [comment]
> @stitesExpensify I am not sure but are you saying that I need to check this https://github.com/Expensify/App/issues/7470#issuecomment-1146848162 ?

Yes, @jayeshmangwani.

**Comment by jayeshmangwani** [comment]
@thesahindia I have tried with safari and chrome and it's working on my side, can you please send me the device/browser where are you getting that this solution is not working?
cc: @mananjadhav @stitesExpensify 
Thanks!
![Screenshot 2022-06-07 at 1 06 20 AM](https://user-images.githubusercontent.com/35371050/172235127-b80f2c6a-262b-437a-95b5-407108ff96d4.png)
![Screenshot 2022-06-07 at 12 55 13 AM](https://user-images.githubusercontent.com/35371050/172235134-036ee071-c481-4d7e-bd97-9649c21ab854.png)
![Screenshot 2022-06-07 at 12 59 19 AM](https://user-images.githubusercontent.com/35371050/172235141-aeffe723-a776-4e28-a7f2-6dd7c7d95b4e.png)
![Screenshot 2022-06-07 at 12 59 29 AM](https://user-images.githubusercontent.com/35371050/172235144-03c15d90-5a6d-4de7-a69a-ea4f9830153a.png)
![Simulator Screen Shot - iPhone 8 - 2022-06-07 at 00 53 28](https://user-images.githubusercontent.com/35371050/172235148-88ef0aa2-ba3d-42ec-9591-32cc51bca59d.png)
![Simulator Screen Shot - iPhone 8 - 2022-06-07 at 00 57 01](https://user-images.githubusercontent.com/35371050/172235155-adecf27a-82b0-42c6-9acb-3107bc3d30b7.png)
![Simulator Screen Shot - iPhone 8 - 2022-06-07 at 01 02 15](https://user-images.githubusercontent.com/35371050/172235159-4b2f847b-75d6-4733-9fec-618f71517c35.png)

**Comment by thesahindia** [comment]
@jayeshmangwani, the background color change is to make it more visible, you also have to use autofill or you need to change the input color so you can see the gap. 

Keep the background red for the parent container and change the `{backgroundColor: 'aliceblue'}` here for the input
https://github.com/Expensify/App/blob/284e0bada7851288be011e2898c91fa81234cc3e/src/components/TextInput/BaseTextInput.js#L259

this way you will see the gap.

**Comment by jayeshmangwani** [comment]
@mananjadhav @stitesExpensify 
I have tried with this [PR](https://github.com/Expensify/App/pull/9294) and I am not getting a white gap at the bottom using autofill, if you get time then can you please check PR once on your side and see the difference in mWeb, I have tried on my side and attached a video for the mWeb in PR too

**Comment by mananjadhav** [comment]
Hmm okay let me test and get back.

@thesahindia Would you mind sharing some more information on your device/browser type for `mWeb` ?

**Comment by thesahindia** [comment]
I tried this on Android chrome. In ios safari it's working fine.

**Comment by trjExpensify** [comment]
@mananjadhav were you able to retest here?

**Comment by mananjadhav** [comment]
Sorry for the delay here @jayeshmangwani. I was OOO earlier and then had to change my machine. 

While the change works as expected for most cases, but as @thesahindia pointed out, it doesn't fix the corners of the input and Android Chrome.

Here are the screenshots for your reference:

<img width="1484" alt="after-the-pr-001" src="https://user-images.githubusercontent.com/3069065/173955517-6807b059-2814-4f70-9693-c8899ed796a8.png">

<img width="361" alt="after-the-pr-002" src="https://user-images.githubusercontent.com/3069065/173955504-764b937a-83e9-4d29-bab0-5da2e1b1ae72.png">


I've zoomed the page on the web to clearly see the gap in the corners and for Android you can use Pixel Android Simulator itself. Can you please add the fix for this? and update once done.

Thank you for your patience.

**Comment by jayeshmangwani** [comment]
@mananjadhav Thanks for testing proposal,
I am trying to solve this issue for android

**Comment by mananjadhav** [comment]
Thank you @jayeshmangwani. And also the corner areas gap for the input.

**Comment by jayeshmangwani** [comment]
@trjExpensify sorry for the delay I have tried with different styling, but I think the root cause for the react-native-web is different, I have tried this issue by creating a new project for react-native-web, but that's working fine. so I am yet trying to figure it out for the web.

**Comment by stitesExpensify** [comment]
Thank you for the update

**Comment by jayeshmangwani** [comment]
@mananjadhav @stitesExpensify 

Updates for the issue,

I tested a few cases with react-native-web and I found that when we wrap a view/textinput in the parent view with the borderWidth then we are getting a slight UI white space around the child, it doesn't matter if we pass the flex or width/height same as a parent there will be little white space around child view, and I am trying from several days and I did not get any success yet, my proposal is not working for the android web then I think we should open this issue for the proposals rather than blocking until I found any solution.

a simple example that I have tested

```
<View
     style={{
            height: 50,
            width: 300,
            borderRadius: 10,
            backgroundColor: 'red',
            borderWidth: 1,
            overflow: 'hidden'
      }}>
          <View style={{backgroundColor: "#fff", flex: 1}} />
  </View>
```

### Android
![Screenshot_20220624-013036_New Expensify](https://user-images.githubusercontent.com/35371050/175387575-cf533f5e-c726-4166-8112-c8fa7fc23d24.jpg)


### Android Chrome
![Screenshot_20220624-013004_Chrome](https://user-images.githubusercontent.com/35371050/175387638-e8694874-2853-49ea-8df6-732644210618.jpg)

in the above example, android took the full height and width of the parent component, but the android web has a small gap at the left, right, and bottom, we can see the red part of the parent component

**Comment by mananjadhav** [comment]
Thanks for the comment @jayeshmangwani. Let me take an opinion internally and get back to you.

**Comment by stitesExpensify** [comment]
Internal Slack thread here https://expensify.slack.com/archives/C02NK2DQWUX/p1656334993341279

**Comment by trjExpensify** [comment]
What are the next steps here? Seems like the consensus in the thread is to proceed with the PR and open a new issue for android mWeb, right?

**Comment by mananjadhav** [comment]
Yeah @trjExpensify. I am going to review this again and test on rest of the platforms.

@jayeshmangwani FYI, based on the internal discussion, we think we should move this task ahead by testing, reviewing and merging this one for all the platforms. For Android mWeb, create a new issue. Can you probably merge the latest `main` branch changes so that we can move this forward.

**Comment by jayeshmangwani** [comment]
@mananjadhav 
Thanks for the update, I have merged the latest main branch and pushed it.

**Comment by mananjadhav** [comment]
Currently Reviewing and retesting it.

**Comment by mananjadhav** [comment]
@jayeshmangwani I did a second round of the tests on all the platforms. I still couldn't find the change where the corners are fixed. Could you please point me to that change? or send me the screenshots of the tests that you performed? I can see those corners in Mac Safari as well as Web Chome (first screenshot in this [comment](https://github.com/Expensify/App/issues/7470#issuecomment-1157026906) )

**Comment by dhairyasenjaliya** [comment]
**Proposal**

- Here actual problem for the white corner border is caused by the border radius of text input as our text input component `<BaseTextInput>` is already wrapped in view that has border-radius of variables.componentBorderRadiusNormal so the simple solution would be to set text input border-radius to 0

- Note: In the latest code (v1.1.82-5) I'm not able to get the top and bottom gap, I'm only able to reproduce all 4 corner visible white gap that was due to text input borderRadius  

**Solution**

- Here set borderRadius: 0, <- It will remove the inner gap of text input make more consistent to parent view

https://github.com/Expensify/App/blob/ce899227ca6d3823e2865a5e194b429f1263db30/src/styles/styles.js#L721-L731
 
**Result**

**Web Chrome without fix can see white corner**
<img width="1011" alt="webChrome-nonFix" src="https://user-images.githubusercontent.com/47522946/178956908-cb88b0c9-9665-41c0-a14b-e159b1770f53.png">

**Web Chrome with fix no visible corner**
<img width="1060" alt="webChromeFix" src="https://user-images.githubusercontent.com/47522946/178956921-2010a876-98cc-4590-8eff-90d1f0a3daf2.png">

**iPhone App without fix**
![iphone-nonfix](https://user-images.githubusercontent.com/47522946/178957587-2e69847b-ac1c-4cea-b80b-9436a48529b6.png)

**iPhone App with fix no visible corner**
![iPhone-fix](https://user-images.githubusercontent.com/47522946/178957599-38e8670e-39a9-49fe-8fe3-de3b76617cfc.png)

**Safari mobile without fix**
![safariMobile-nonFix](https://user-images.githubusercontent.com/47522946/178957807-8a71d766-ea4e-4285-8492-46de0a938799.png)

**Safari mobile with fix no visible corner**
![safariMobile-Fix](https://user-images.githubusercontent.com/47522946/178957790-1bac7671-4e6e-45fd-817c-4e0175e97f1d.png) 

**Safari web without fix**
<img width="834" alt="Mac-Safari-nonFix" src="https://user-images.githubusercontent.com/47522946/178958308-e29ec524-1150-4d99-8439-bac9b62a7dc1.png">

**Safari web with fix no visible corner**
<img width="932" alt="Mac-Safari-Fix" src="https://user-images.githubusercontent.com/47522946/178958321-3a070b74-e031-49b6-9693-daebb919e4b2.png">

**Android App without fix**
![Android-nonFix](https://user-images.githubusercontent.com/47522946/178958123-cb46b7e2-d7ab-42c2-93a1-deb4b8da257a.jpg)

**Android App with fix no visible corner**
![Android-fix](https://user-images.githubusercontent.com/47522946/178958133-aebe38ea-2f1e-452a-8e64-77f11e839604.jpg)

**Chrome mobile without fix**
![ChromeAndroid-nonFix](https://user-images.githubusercontent.com/47522946/178958531-ca6b1a76-988f-4dcd-9139-c799e9e23c62.jpg)

**Chrome mobile with fix no visible corner but I can still see tiny top & bottom white gap ** 
![ChromeAndroid-Fix](https://user-images.githubusercontent.com/47522946/178958539-b8e56729-292e-43d6-b527-8c2f94f902cc.jpg)

**Comment by jayeshmangwani** [comment]
@mananjadhav Thanks for testing PR, yes I have not added code for the border, as this was not the issue of the height, but I found that this issue is happening because we are adding borderRadius to child and parent both, and in that scenario, the child is not taking the full height/width and we can see 1 pixel of parent's view, so I would suggest that we can remove the child's(TextInput) borderRadius because we have already borderRadius and borderWidth to the parent component

Here we are passing the borderRadius
https://github.com/Expensify/App/blob/8718a905ce72ae1b3f9336c0e6bc1bf920e650f9/src/components/TextInput/BaseTextInput.js#L215
https://github.com/Expensify/App/blob/8718a905ce72ae1b3f9336c0e6bc1bf920e650f9/src/styles/styles.js#L759-L773

so, I am suggesting we should remove borderRadius from here

https://github.com/Expensify/App/blob/8718a905ce72ae1b3f9336c0e6bc1bf920e650f9/src/styles/styles.js#L721-L731
```diff
-       borderRadius: variables.componentBorderRadiusNormal,
```

this is a basic example for explaining the above behavior,
here we have 2 view components with the same height and width, and parent has borderwidth 1 and child has borderwidth 0, so technically child black color should cover the entire view, but if you check this you will see a blue pixel from the parent

```
<View style={{
        height: 50,
        width: 100,
        borderRadius: 10,
        backgroundColor: 'blue',
        borderWidth: 1,
        borderColor: "red",
        overflow: "hidden",
     }}>
        <View style={{
            height: 50,
            width: 100,
            borderRadius: 10,
            backgroundColor: 'black',
            borderWidth: 0,
        }} />
</View>
```

**Comment by mananjadhav** [comment]
@jayeshmangwani The change looks fine, can you please add and update the PR?

**Comment by dhairyasenjaliya** [comment]
@mananjadhav can you check my proposal as I identify the actual root cause and provide a solution very first https://github.com/Expensify/App/issues/7470#issuecomment-1184258850

**Comment by thesahindia** [comment]
@jayeshmangwani is already assigned here. @dhairyasenjaliya, search other issues with the help wanted label.

**Comment by dhairyasenjaliya** [comment]
yeah can understand :)

**Comment by jayeshmangwani** [comment]
@mananjadhav PR updates, Thanks!

**Comment by mananjadhav** [comment]
@jayeshmangwani Will test today.

**Comment by trjExpensify** [comment]
[PR hit staging](https://github.com/Expensify/App/pull/9294) a couple of hours ago, Melv.

**Comment by marcaaron** [comment]
Unfortunately, there was a regression caused by this PR and it is being reverted now.

**Comment by mananjadhav** [comment]
@marcaaron can you link the issue where we are tracking the regression ? I can also take a look

**Comment by mananjadhav** [comment]
@jayeshmangwani We'll have to check the regression and fix this. Its happening because onLayout is taking the full height of the View.

**Comment by jayeshmangwani** [comment]
> @jayeshmangwani We'll have to check the regression and fix this. Its happening because onLayout is taking the full height of the View. 

@mananjadhav Okay I will check this regression in 2-3 days I am not working few days so , please can we wait next 3 days, Thanks

**Comment by jayeshmangwani** [comment]
@mananjadhav Thanks for waiting on this regression,

I have checked this issue and found that at WorkspaceInvitePage this issue is happening because we are passing minHeight style to containerStyles, if we are passing minHeight to containerStyles then onLayout calling constantly with a new height every time,

and I think 2 ways to handle this issue, 

1. we will change minHeight to height, as we are already passing numberOfLines in the WorkspaceInvitePage TextInput so height is always bound in 4 lines

https://github.com/Expensify/App/blob/a06358cef3ab1951fcbb5c256c003acdf238d447/src/pages/workspace/WorkspaceInvitePage.js#L297-L307

https://github.com/Expensify/App/blob/a06358cef3ab1951fcbb5c256c003acdf238d447/src/styles/styles.js#L2343-L2345

```diff
+height: 115
-minHeight: 115
```

2. we are getting [issue](https://github.com/Expensify/App/issues/7470) in the autofill , and autofill always will be in single line, so when we pass multiline prop to TextInput then no need to add onlayout and dynamic height,

please let me know your thoughts on the above 2 proposed solutions

**Comment by jayeshmangwani** [comment]
@mananjadhav 
Hey, have you got a chance to check my [proposed](https://github.com/Expensify/App/issues/7470#issuecomment-1200724434) changes for the regression?

**Comment by mananjadhav** [comment]
@jayeshmangwani I did check. While the prop thing looks good, would you mind checking one change for me? what if we remove the `onLayout` code and only retain the border change. Do you think that'll work?

**Comment by jayeshmangwani** [comment]
@mananjadhav checked without `onLayout` but a white gap is visible,

without onLayout code
![Screenshot_20220810-014558_New Expensify](https://user-images.githubusercontent.com/35371050/183753462-1770380a-fd5b-43a1-ae04-4f869be46109.jpg)

with onLayout code
![Screenshot_20220810-014649_New Expensify](https://user-images.githubusercontent.com/35371050/183753550-62fbe70a-d70d-4d43-8fd1-655a75b1094f.jpg)

**Comment by trjExpensify** [comment]
This was reverted, so I'm removing the payment hold from the title and the label.

**Comment by jayeshmangwani** [comment]
@mananjadhav these are the changes we will need to make,

https://github.com/Expensify/App/blob/e57e113e8ead6ae7ad0068c91593504dc101b160/src/components/TextInput/BaseTextInput.js#L214-L221

```diff
+{...!this.props.multiline ? {
+  onLayout: event =>
+      this.setState({height: event.nativeEvent.layout.height})
+  } : {}
+}
```

https://github.com/Expensify/App/blob/e57e113e8ead6ae7ad0068c91593504dc101b160/src/components/TextInput/BaseTextInput.js#L260-L267
```diff
+!this.props.multiline && {height: this.state.height},
```

https://github.com/Expensify/App/blob/e57e113e8ead6ae7ad0068c91593504dc101b160/src/styles/styles.js#L731-L741
```diff
- borderRadius: variables.componentBorderRadiusNormal,
```

it will cover all the single line inputs

**Comment by mananjadhav** [comment]
Thanks for the updated proposal @jayeshmangwani. This one should work. @stitesExpensify anything you'd like to check?

**Comment by stitesExpensify** [comment]
Looks good to me!

**Comment by mananjadhav** [comment]
@jayeshmangwani Can you submit the PR with the updated fix?

**Comment by jayeshmangwani** [comment]
@mananjadhav PR raised
cc: @stitesExpensify

**Comment by mananjadhav** [comment]
@jayeshmangwani for this one do you mind uploading a video of Web and 1 Mobile platform that covers the whole app different screens just to be sure we're not breaking anything? I'll take a look at the PR after that.

**Comment by jayeshmangwani** [comment]
@mananjadhav sure , I am away from work for 2 days , I will upload on Tuesday

**Comment by jayeshmangwani** [comment]
@mananjadhav I am uploading videos for web and android to Gdrive due to video size, please check these updated videos

web
https://drive.google.com/file/d/1zfoyvIIqQ4kMSXrf49VA_Ws6lcbBvRby/view?usp=sharing

Android
https://drive.google.com/file/d/1IyJItxEOku422YYFOdHxDQpGny-QqNgU/view?usp=sharing

**Comment by mananjadhav** [comment]
quick update, I am testing this (also while generally using the app), to ensure that we don't break anything. It should be ready to merge in this week.

**Comment by mananjadhav** [comment]
PR under review and in final stages of approval.

**Comment by trjExpensify** [comment]
👋 Alrighty, time to settle up! To confirm.. 

$250 - @rushatgabhane for reporting 
$4,000 - @jayeshmangwani for the fix 
$2,000 - @mananjadhav for C+ (1/2 because of the revert [here](https://github.com/Expensify/App/issues/7470#issuecomment-1210592772))

Let me know if that's accurate!

**Comment by mananjadhav** [comment]
Sounds right to me, @trjExpensify

**Comment by trjExpensify** [comment]
Alrighty, let's do this! 

- @jayeshmangwani settled up! 
- @rushatgabhane - contract sent 
- @mananjadhav - contract sent

**Comment by jayeshmangwani** [comment]
@trjExpensify Thanks!

**Comment by mananjadhav** [comment]
Accepted @trjExpensify
